### DISCUSSION TITLE
Installazione su rasp 5: alcuni dubbi

### DISCUSSION THREAD

**Discussion by crug80** [indomus/forum#2780] [HUB: Home Assistant]
Salve a tutti, vorrei avere alcuni chiarimenti sul tema installazione. Ho letto le guide e capito le varie sfumature circa le differenze, così vorrei un vostro parere. 
Attualmente ho un'installazione di prova in virtualBox (HassOS) che sta funzionando bene e che rappresentava la soluzione temporanea in vista del nuovo Rasp 5.  Durante questi giorni ho provato anche una rasp 3 b+, ma ovviamente l'estrema lentezza mi ha convinto a lasciar perdere. L'opzione in uso funziona bene, ma mi costringe a schedulare riavvii per svuotare la memoria, oppure soluzioni tampone per l'hardware virtualizzato che ho installato (convertitori usb rs485). 
Oggi mi è arrivata la nuova rasp 5 8Gb, ma leggendo sul thread ufficiale di HassOS circa il porting verso questa nuova board ho capito che sono ancora in dev, ma a me serve una soluzione stabile. 
Pertanto, partendo dal presupposto che devo importare il backup completo del mio sistema attuale (componenti da repository di terze parti non ufficiali, HACS, etc. etc.) da ciò che ho capito dalle vostre guide l'unico modo per portarmi dietro il tutto è installare la modalità supervised. 
Corretto? 
Ora, se è corretto (se non lo è, correggetemi), ci sono solo 2 OS al momento che ufficialmente supportano la Rasp 5: Rasp OS e Ubuntu. 
Voi che fareste? 
Aspettereste ancora, oppure provereste ad installarlo sul Rasp OS con docker?!

P.S:
Vi segnalo che sul vostro c'è questa cosa molto fuorviante:
<img width="580" alt="Rasp_OS" src="https://github.com/indomus/forum/assets/41435902/3d5a45b7-6a8b-46d0-a066-283b797b5d38">
In realtà l'articolo a cui punta il link è quello di installazione su Debian e non Rasp OS

**Comment by snakuzzo** [comment]
Nessuna delle due è ufficialmente supportata per il tipo di installazione HA Supervised.
IMHO, salvo casi eccezionali, hai sbagliato a comprare il RPI5. Potevi prendere uno dei tanti miniPC in commercio...lo consigliamo da anni ormai.
Comunque, se proprio vuoi un'installazione sulla quale possa essere ripristinato il tuo backup "paro paro", o aspetti Debian per RPI5 e installi HA Supervised, oppure aspetti la release di HA OS per RPI5.
Se proprio non ti va di aspettare, impara docker e ti porti man mano tutto quello che hai.

**Reply by crug80** [reply]
Diciamo che difficilmente avrei trovato un mini pc ad 80€ con consumi bassi come il Rasp 5 e con il supporto per installazione su barra din.. 
Intanto grazie per le precisazioni. Probabilmente aspetterò qualche giorno e magari anche il testing dell'hat per la ssd m.2.. 

Ma giusto per curiosità, come funziona il meccanismo degli add-on e dei backup sulla Core?

**Reply by snakuzzo** [reply]
Sul discorso rpi e barra din e tutto il resto mi taccio, altrimenti andiamo a finire sempre sui soliti discorsi e si rischia di discutere inutilmente.

Quanto agli addon, installando core dovresti creare a manella i vari container docker relativi alle applicazioni che solitamente trovi come addon...e comunque non c'è un modo di ripristinare in automatico le configurazioni di un addon. Dovresti metterti li a manella, studiare come è fatta l'immagine docker, creare il container secondo indicazioni e riportare le varie configurazioni

**Reply by crug80** [reply]
Tranquillo non è mia intenzione né fare polemica o altro. Non conosco "i soliti discorsi" essendo nuovo di HA e di questo forum.. 
Detto questo, pur se per curiosità mi ci metterei pure a smanettare, al momento ho esigenze di concretezza e rapidità, quindi aspetterò qualcosa di ufficiale continuando a lavorare intanto in virtuale.. 
[OT]
Visto che sei un "collaborator" e sicuramente più esperto di me, sono sfacciato o infrango qualche regole se ti chiedo una cortesia, ovvero di dare uno sguardo al mio post sui command_line?! (son 3 giorni che sto fermo sul problema senza via d'uscita)

**Comment by crug80** [comment]
Oggi è uscito l'aggiornamento del supervisor con il supporto per Pi5..

**Reply by KC-inDomus** [reply]
sì, ma solo il supervisor, per ora.

**Comment by Bergoglio** [comment]
Come meglio credi, ma personalmente trovo che una installazione Hass OS su di un mini PC è insuperabile. È vero, un mini pc valido vuol dire 200 euro contro 80, ma quei 120 di differenza secondo me vanno via velocemente in medicine per il fegato......
Io valuterei il passaggio al mini pc, anche se hai già in casa il Pi5....

**Reply by crug80** [reply]
Al di là dei costi, il mio problema è la barra din con poco spazio. Ho il cablaggio del modbus che esce all'interno del quadro e portarlo fuori in modo elegante è quasi impossibile. Lo testo, poi mi regolo. Se l'esperienza non sarà fluida lo riuso per altro e mi muovo su un hardware diverso inventandomi qualcosa per far uscire il cavo.

**Reply by Bergoglio** [reply]
E se mettessi un convertitore Modbus > TCP/IP ?
Io ho montato questi https://www.waveshare.com/product/iot-communication/wired-comm-converter/ethernet-to-uart-rs232-rs485/rs485-to-eth-b.htm comprati su amazon, ma questo https://www.waveshare.com/product/iot-communication/wired-comm-converter/ethernet-to-uart-rs232-rs485/rs485-to-wifi-eth.htm e questo https://www.waveshare.com/product/iot-communication/wired-comm-converter/ethernet-to-uart-rs232-rs485/rs232-485-to-wifi-eth-b.htm addirittura sono wifi.

**Reply by crug80** [reply]
Giusto! Cha idiota, ero così focalizzato sul problema cavo che la possibilità del gateway l'avevo proprio rimossa dalla mia testa. Grazie, ora il piano b è decisamente più facile. 
Lo so, son cocciuto, ma ormai il piano A lo voglio testare come va :)

**Comment by crug80** [comment]
Aggiornamenti: installata la mia rasp 5 con SD card e migrato il tutto con il backup. Attualmente (con i dovuti scongiuri) sta girando bene. Esperienza molto fluida e nei vari test mai impegnato il processore per più del 5% e RAM più dei 2Gb sugli 8 disponibili. 
Appena stabilizzano il discorso boot da ssd, provo a migrare con hat dedicato su ssd..
